packages/shortest/tests/unit/tools/index.test.ts

```diff
@@ -0,0 +1,128 @@
+
+      const mockBrowserTool = {
+        execute: vi.fn().mockResolvedValue({}),
+        resultToToolResultContent: vi.fn(),
+      } as unknown as BrowserTool;
+
+      const tools = registry.getTools(
+        "anthropic",
+        "claude-3-5-sonnet-latest",
+        mockBrowserTool,
+      );
+
+      expect(tools).toHaveProperty("computer");
+      expect(tools).toHaveProperty("bash");
+      expect(tools).toHaveProperty("check_email");
+      expect(tools).toHaveProperty("github_login");
+      expect(tools).toHaveProperty("navigate");
+      expect(tools).toHaveProperty("run_callback");
+      expect(tools).toHaveProperty("sleep");
+
+      const expectedToolCount = 7; // 2 provider tools + 5 custom tools
+      expect(Object.keys(tools).length).toBe(expectedToolCount);
+
+      const toolsMap = (registry as any).tools as Map<string, any>;
+
+      const computerTool = toolsMap.get("anthropic_computer_20241022");
+      expect(computerTool).toEqual({
+        name: "computer",
+        category: "provider",
+        factory: createAnthropicComputer20241022,
+      });
+
+      const bashTool = toolsMap.get("anthropic_bash_20241022");
+      expect(bashTool).toEqual({
+        name: "bash",
+        category: "provider",
+        factory: createAnthropicBash20241022,
+      });
+
+      const checkEmailTool = toolsMap.get("check_email");
+      expect(checkEmailTool).toEqual({
+        name: "check_email",
+        category: "custom",
+        factory: createCheckEmailTool,
+      });
+
+      const githubLoginTool = toolsMap.get("github_login");
+      expect(githubLoginTool).toEqual({
+        name: "github_login",
+        category: "custom",
+        factory: createGithubLoginTool,
+      });
+
+      const navigateTool = toolsMap.get("navigate");
+      expect(navigateTool).toEqual({
+        name: "navigate",
+        category: "custom",
+        factory: createNavigateTool,
+      });
+
+      const runCallbackTool = toolsMap.get("run_callback");
+      expect(runCallbackTool).toEqual({
+        name: "run_callback",
+        category: "custom",
+        factory: createRunCallbackTool,
+      });
+
+      const sleepTool = toolsMap.get("sleep");
+      expect(sleepTool).toEqual({
+        name: "sleep",
+        category: "custom",
+        factory: createSleepTool,
+      });
+    });
+
+    it("provides tools with execute functionality", () => {
+      const registry = createToolRegistry();
+
+      const mockBrowserTool = {
+        execute: vi.fn().mockResolvedValue({}),
+        resultToToolResultContent: vi.fn(),
+      } as unknown as BrowserTool;
+
+      const tools = registry.getTools(
+        "anthropic",
+        "claude-3-5-sonnet-latest",
+        mockBrowserTool,
+      );
+
+      Object.values(tools).forEach((tool) => {
+        expect(tool).toHaveProperty("execute");
+
+        expect(typeof tool.execute).toBe("function");
+      });
+    });
+
+    it("registers tools with expected keys", () => {
```

Comment:
This ensures all the tools available are properly registered. If a tool doesn't get registered, and not passed to the AI, the AI will do its best to complete the task with the missing tool. It will not be obvious - through a script error - that a tool is missing).